### PR TITLE
Added WithAggregationSelector to prometheus

### PR DESCRIPTION
**Pull request by MadVikingGod**
Fixes: #3196

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 ### Added
 
 - Prometheus exporter will register with a prometheus registerer on creation, there are options to control this. (#3239)
+- Added an option to change the AggregationSelector for prometheus. (#????)
```
Comment (MrAlias) [inline CHANGELOG.md:14]:
```suggestion
- Added the `WithAggregationSelector` option to the `go.opentelemetry.io/otel/exporters/prometheus` package to change the `AggregationSelector` used. (#3341)
```

File: exporters/prometheus/confg_test.go [hunk 2]
```diff
@@ -58,3 +80,33 @@ func TestNewConfig(t *testing.T) {
...
 	}
 }
+
+func TestConfigManualReaderOptions(t *testing.T) {
+	aggregationSelector := func(view.InstrumentKind) aggregation.Aggregation { return nil }
+
+	testCases := []struct {
+		name       string
+		config     config
+		wantOption []metric.ManualReaderOption
```
Comment (MrAlias) [inline exporters/prometheus/confg_test.go:90]: This does not test the value of these options. Is there any reason this doesn't instead accept just an expected option length here?

File: exporters/prometheus/config.go [hunk 3]
```diff
@@ -57,3 +67,10 @@ func WithRegisterer(reg prometheus.Registerer) Option {
 		return cfg
 	})
 }
+
+func WithAggregationSelector(agg metric.AggregationSelector) Option {
```
Comment (MrAlias) [inline exporters/prometheus/config.go:71]: This is an exported function, it needs to be documented.

File: exporters/prometheus/confg_test.go (hunk 2 above)
Comment (MadVikingGod) [reply exporters/prometheus/confg_test.go:90]: I did this to try and compare equality, but I haven't been able to figure out how.  I think for now I will move it to count and roll it into the test above.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**